File: mobile/lib/features/department/domain/entities/general_department.dart
import 'package:equatable/equatable.dart';

import 'department.dart';

class GeneralDepartment extends Equatable {
  final String id;
  final String name;
  final String description;
  final List<Department> departments;
  final bool isForListing;

  const GeneralDepartment({
    required this.id,
    required this.name,
    required this.description,
    required this.departments,
    required this.isForListing,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        description,
        isForListing,
      ];
}

File: new_backend/src/models/generalDepartment.ts
import { Document, Schema, model } from "mongoose";

export interface IGeneralDepartment extends Document {
//   id: Schema.Types.ObjectId;
  description: String;
  isForListing: Boolean
  name: String;
 
}

const GeneralDepartmentSchema: Schema<IGeneralDepartment> = new Schema({
  name:{
    type: String,
    required:[true,"General department name is required"]
  },
  isForListing: {
    type: Boolean,
    default: true
  },
  description: {
    type: String,
    required: false,
    default: ""
  }
},
{
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt'
    }
  }
);

const GeneralDepartment = model<IGeneralDepartment>("GeneralDepartment", GeneralDepartmentSchema);

export default GeneralDepartment;
